app/controllers/application_controller.rb

```diff
@@ -2,4 +2,59 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_action :current_user
+  before_action :current_order
+  before_action :items_in_cart
+
+
+  def current_order
+    if session[:order_id] && !!Order.exists?(session[:order_id])
```

Comment:
What's the reason for using double bang here (and other places)? `.exists?` already returns a boolean value.